### ISSUE TITLE
Avoid Overlap not working in certain instance of Reshape feature 3.16.5 and 3.10.11

### ISSUE THREAD

**Issue by LucySaluki** [qgis/QGIS#42595]
Avoid Overlap is not working as I am expecting in certain circumstance when I use reshape feature.  I have three polygons (capture 1) and wish to reshape the blue one.  If I reshape it but do not touch the other neighbouring polygon (capture 2 and 3) the it all works as expected.  If however I snap to another neighbouring polygon (green) then the avoid overlap doesnt work (capture 4 and 5). As I have just downloaded Hannover I have tested this in both 3.10 A Coruna and 3.16 Hannover and I get the same.  Thanks
![Capture5](https://user-images.githubusercontent.com/46525147/113124426-6a43f200-920d-11eb-8fdd-315fcce868cb.JPG)
![Capture4](https://user-images.githubusercontent.com/46525147/113124435-6c0db580-920d-11eb-9f15-834fab7d35f4.JPG)
![Capture3](https://user-images.githubusercontent.com/46525147/113124440-6d3ee280-920d-11eb-94ab-69687870f000.JPG)
![Capture2](https://user-images.githubusercontent.com/46525147/113124448-6e700f80-920d-11eb-88bc-9c09c0ddea1f.JPG)
![Capture1](https://user-images.githubusercontent.com/46525147/113124456-7039d300-920d-11eb-995e-5148194883b6.JPG)

**Comment by lbartoletti** [comment]
Should be fixed in https://github.com/qgis/QGIS/pull/41658

**Comment by uclaros** [comment]
@lbartoletti 
> Should be fixed in #41658

No, it should not, this is about the reshape tool, not the vertex tool.

@LucySaluki What you experience here has nothing to do with *Avoid overlaps*, you may disable it as well. The *Reshape tool* tries to reshape all the features intersecting your digitized line, so in your first case both the red and blue polygons get reshaped.
On your second case though, the resulting geometry from reshaping the red polygon is *invalid* because its inner and outer rings intersect, so only the blue polygon gets reshaped instead. Some warning message is obviously missing here...

**Comment by gioman** [comment]
> On your second case though, the resulting geometry from reshaping the red polygon is _invalid_ because its inner and outer rings intersect, so only the blue polygon gets reshaped instead. Some warning message is obviously missing here...

@uclaros so better chamge the title to something along the lines of "missing warning message in reshape tool"?

**Comment by uclaros** [comment]
@gioman Yes, that would make sense for me!

**Comment by LucySaluki** [comment]
Ok thanks, I understand, and yes a warning message would be helpful. Feel free to close the issue if you need to.
